lib/resources/nginx_conf.rb

```diff
@@ -93,13 +96,33 @@ def resolve_references(data, rel_path)
       if data.key?('include')
         data.delete('include').flatten
             .map { |x| File.expand_path(x, rel_path) }
+            .map { |x| find_files(x) }.flatten
             .map { |path| parse_nginx(path) }
-            .map { |e| data.merge!(e) }
+            .each { |conf| merge_config!(data, conf) }
       end
 
       # Walk through the remaining hash fields to find more references
       Hash[data.map { |k, v| [k, resolve_references(v, rel_path)] }]
     end
+
+    # Deep merge fields from NginxConfig.parse.
+    # A regular merge would overwrite values so a deep merge is needed.
+    # @param data [Hash] data structure from NginxConfig.parse
+    # @param conf [Hash] data structure to be deep merged into data
+    # @return [Hash] data structure with conf and data deep merged
+    def merge_config!(data, conf)
+      # Catch edge-cases
+      return if data.nil? || conf.nil?
+      # Step through all conf items and create combined return value
+      data.merge!(conf) do |_, v1, v2|
+        # If both the data field and the conf field are arrays, then combine them
+        next v1 + v2 if v1.is_a?(Array) && v2.is_a?(Array)
```

Comment:
Maybe it's just me, but this use of `next` feels pretty awkward. Any chance I can convince you guys to make this an `if ... elsif ... else` instead? I think it would read so much cleaner.